include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_consistent_splitting.cpp

```diff
@@ -0,0 +1,895 @@
+                            this->adaptive_time_stepping,
+                            this->get_time_step_vector());
+
+  // use this function to check the correctness of the time integrator constants
+  //    std::cout << "Coefficients extrapolation scheme pressure NBC:" << std::endl;
+  //    extra_pressure_nbc.print(this->pcout);
+}
+
+template<int dim, typename Number>
+void
+TimeIntBDFConsistentSplitting<dim, Number>::setup_derived()
+{
+  Base::setup_derived();
+
+  // velocity_dbc vectors do not have to be initialized in case of a restart, where
+  // the vectors are read from restart files.
+  if(this->param.restarted_simulation == false)
+  {
+    initialize_velocity_dbc();
+  }
+}
+
+template<int dim, typename Number>
+void
+TimeIntBDFConsistentSplitting<dim, Number>::allocate_vectors()
+{
+  Base::allocate_vectors();
+
+  // velocity
+  pde_operator->initialize_vector_velocity(velocity_np);
+  for(unsigned int i = 0; i < velocity.size(); ++i)
+    pde_operator->initialize_vector_velocity(velocity[i]);
+
+  // pressure
+  pde_operator->initialize_vector_pressure(pressure_np);
+  for(unsigned int i = 0; i < pressure.size(); ++i)
+    pde_operator->initialize_vector_pressure(pressure[i]);
+
+  // velocity divergence
+  for(unsigned int i = 0; i < velocity_divergence.size(); ++i)
+    pde_operator->initialize_vector_pressure(velocity_divergence[i]);
+
+  // convective term divergence
+  for(unsigned int i = 0; i < vec_convective_term_div.size(); ++i)
+    pde_operator->initialize_vector_pressure(vec_convective_term_div[i]);
+
+  // velocity_dbc
+  pde_operator->initialize_vector_velocity(velocity_dbc_np);
+  for(unsigned int i = 0; i < velocity_dbc.size(); ++i)
+    pde_operator->initialize_vector_velocity(velocity_dbc[i]);
+}
+
+
+template<int dim, typename Number>
+void
+TimeIntBDFConsistentSplitting<dim, Number>::initialize_current_solution()
+{
+  pde_operator->prescribe_initial_conditions(velocity[0], pressure[0], this->get_time());
+
+  // Now compute divergence of velocity and convective term
+  VectorType velocity_divergence_np(pressure[0]);
+  velocity_divergence_np = 0.;
+  pde_operator->compute_Leray_projection_term(velocity_divergence_np,
+                                              velocity[0],
+                                              this->get_time());
+  velocity_divergence[0].swap(velocity_divergence_np);
+
+  VectorType vec_convective_term_div_np(pressure[0]);
+  vec_convective_term_div_np = 0.;
+  pde_operator->apply_convective_divergence_term(vec_convective_term_div_np,
+                                                 velocity[0],
+                                                 this->get_time());
+  vec_convective_term_div[0].swap(vec_convective_term_div_np);
+}
+
+template<int dim, typename Number>
+void
+TimeIntBDFConsistentSplitting<dim, Number>::initialize_former_multistep_dof_vectors()
+{
+  // Note that the loop begins with i=1, we could also start with i=0 but this is not necessary.
```

Comment:
Can we remove the second part this comment, I don't think it contributes to the understanding. The reason is that the data for the time value 0 is set elsewhere.